modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb

```diff
@@ -21,6 +21,7 @@ def self.date_from_string(string)
     attr_encrypted(:auth_headers, key: Settings.db_encryption_key, marshal: true, marshaler: JsonMarshal::Marshaller)
 
     validate :validate_hearing_type_selection
+    validate :validate_address_unless_homeless, on: :create
```

Comment:
Reminder to move the validations to a `validate` method with an `if:` proc like HLR does.